### ISSUE TITLE
Apple Mach-O Linker Error

### ISSUE THREAD

**Issue by chasedson** [kpwn/yalu102#468]
I read the issue posting guidelines.

Undefined symbols for architecture i386:
  "_whole_dump", referenced from:
      _exploit in jailbreak.o
  "_whole_base", referenced from:
      _exploit in jailbreak.o
  "_text_exec_base", referenced from:
      _exploit in jailbreak.o

Basically that error when building, but a long longer. What is going wrong?

**Comment by ghost** [comment]
are u building for 32 or 64 bit, this is only 64bit jailbreak so use arm64 to build not armv7

**Comment by Mila432** [comment]
@ByteGig wow you really know your stuff ..

**Comment by ghost** [comment]
me i dont even have a mac so i cant build it, i teach myself stuff Twitter @ByteGig and I don't even know how i got here! Please help me btw reach my future hacking career. my website is gig-a-byte.co.nf

**Comment by jakeajames** [comment]
@ByteGig You're wrong. yalu102 is a armv7 only application which only works on 64 bit devices. Doesn't make too much sense but that's the truth

**Comment by ghost** [comment]
@jakeajames i am not wrong yalu102 is armv7 but ian beers exploit can be ported to 32bit and so can armv7. U need to port it which means changing the code so it is 32bit but it takes a while. and u r wrong armv7 is 32bit #makearmv7greatagain

**Comment by jakeajames** [comment]
@ByteGig You said "so use arm64 to build". That's WRONG. It will NEVER build at arm64.

**Comment by endercypher** [comment]
@jakeajame @ByteGig yes you are correct @jakeajames it won't ever build at Arm64 it will only build as Armv7 because the Kernel exploit has to run as 32-Bit to bypass KPP which is for 64-BIT DEVICES ONLY I DONT EVEN HAVE REMOTE ACCESS TO A MAC AND I KNOW MORE THAN YOU!
